airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,468 @@
+                    'shell': 'true',
+                    'environment': {
+                        'variables': [
+                            {
+                                'name':'AIRFLOW__CORE__SQL_ALCHEMY_CONN',
+                                'value': self.core_sql_alchemy_conn
+                            },
+                            {
+                                'name':'AIRFLOW__CORE__FERNET_KEY',
+                                'value': self.core_fernet_key
+                            },
+                            {
+                                'name':'AIRFLOW__CORE__LOGGING_LEVEL',
+                                'value': 'DEBUG'
+                            }
+                        ]
+                    },
+                    'value': " ".join(cmd)
+                },
+                'container': {
+                    'type': 'DOCKER',
+                    'volumes' : [
+                        {
+                            'container_path' : self.mesos_docker_volume_dag_container_path,
+                            'mode' : 'RO',
+                            'source' : {
+                                'type' : 'DOCKER_VOLUME',
+                                'docker_volume' : {
+                                    'driver': self.mesos_docker_volume_dag_driver,
+                                    'name' : self.mesos_docker_volume_dag_name
+                                }
+                            }
+                        }
+                    ],
+                    'docker': {
+                        'image': self.mesos_slave_docker_image,
+                        'force_pull_image': 'true',
+                        'privileged' : 'true',
+                        'parameters' : [
+                            {
+                                'key' : 'volume',
+                                'value' : self.mesos_docker_sock + ':/var/run/docker.sock'
+                            }
+                        ]
+                    }
+                }
+            }
+
+            option = {
+                'Filters' : {
+                    'RefuseSeconds': '0.5'
+                }
+            }
+
+            tasks.append(task)
+            remaining_cpus -= self.task_cpu
+            remaining_mem -= self.task_mem
+        mesos_offer.accept(tasks, option)
+
+    @provide_session
+    def subscribed(self, driver, session=None):
+        """
+        Subscribe to Mesos Master
+
+        :param driver: Mesos driver object
+        """
+        from airflow.models import Connection
+
+        # Update the Framework ID in the database.
+        conn_id = FRAMEWORK_CONNID_PREFIX + get_framework_name()
+        connection = session.query(Connection).filter_by(conn_id=conn_id).first()
+        if connection is None:
+            connection = Connection(
+                conn_id=conn_id,
+                conn_type='mesos_framework-id',
+                extra=driver.frameworkId
+            )
```

Comment:
This does not create a connection in database, is it intended?

Reply:
Thats a piece of the old mesos_executor. It should update the framework_id in the database. And as I saw, it was working. If thats not the way how to do it, please tell how I should do it better.